go/test/endtoend/tabletmanager/zhealth_check_test.go

```diff
@@ -0,0 +1,396 @@
+	if err != nil {
+		t.Fatal(err)
+	}
+	serving := streamHealthResponse.GetServing()
+	UID := streamHealthResponse.GetTabletAlias().GetUid()
+	realTimeStats := streamHealthResponse.GetRealtimeStats()
+	secondsBehindMaster := realTimeStats.GetSecondsBehindMaster()
+	fmt.Println("**********", secondsBehindMaster)
+	assert.True(t, serving, "Tablet should be in serving state")
+	// assert.True(t, secondsBehindMaster < 30, "Slave should not be behind master")
+	assert.True(t, UID > 0, "Tablet should contain uid")
+}
+
+func TestHealthCheckDrainedStateDoesNotShutdownQueryService(t *testing.T) {
+	// This test is similar to test_health_check, but has the following differences:
+	// - the second tablet is an 'rdonly' and not a 'replica'
+	// - the second tablet will be set to 'drained' and we expect that
+	// - the query service won't be shutdown
+
+	//Wait if tablet is not in service state
+	waitForTabletStatus(rdonlyTablet, "SERVING")
+
+	// Check tablet health
+	checkHealth(t, rdonlyTablet.HTTPPort, false)
+	assert.Equal(t, "SERVING", rdonlyTablet.VttabletProcess.GetTabletStatus())
+
+	// Change from rdonly to drained and stop replication. (These
+	// actions are similar to the SplitClone vtworker command
+	// implementation.)  The tablet will stay healthy, and the
+	// query service is still running.
+	err := clusterInstance.VtctlclientProcess.ExecuteCommand("ChangeSlaveType", rdonlyTablet.TabletAlias, "drained")
+	assert.Nil(t, err, "error should be Nil")
+	// Trying to drain the same tablet again, should error
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("ChangeSlaveType", rdonlyTablet.TabletAlias, "drained")
+	assert.Error(t, err, "already drained")
+
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("StopSlave", rdonlyTablet.TabletAlias)
+	assert.Nil(t, err, "error should be Nil")
+	// Trigger healthcheck explicitly to avoid waiting for the next interval.
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("RunHealthCheck", rdonlyTablet.TabletAlias)
+	assert.Nil(t, err, "error should be Nil")
+
+	result, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("GetTablet", rdonlyTablet.TabletAlias)
+	assert.Nil(t, err, "error should be Nil")
+	checkTabletType(t, result, "DRAINED")
+
+	// Query service is still running.
+	waitForTabletStatus(rdonlyTablet, "SERVING")
+
+	// Restart replication. Tablet will become healthy again.
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("ChangeSlaveType", rdonlyTablet.TabletAlias, "rdonly")
+	assert.Nil(t, err, "error should be Nil")
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("StartSlave", rdonlyTablet.TabletAlias)
+	assert.Nil(t, err, "error should be Nil")
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("RunHealthCheck", rdonlyTablet.TabletAlias)
+	assert.Nil(t, err, "error should be Nil")
+	checkHealth(t, rdonlyTablet.HTTPPort, false)
+}
+
+func waitForTabletStatus(tablet cluster.Vttablet, status string) {
```

Comment:
This can be moved to vttablet_process , by default it should check for NOT_SERVING and from other places it will check for the status being passed over parameter.

Reply:
Done